packages/next/src/server/base-server.ts

```diff
@@ -2144,27 +2135,20 @@ export default abstract class Server<
       }
     }
 
-    if (isAppPath) {
-      if (!this.renderOpts.dev && !isPreviewMode && isSSG && isRSCRequest) {
-        // If this is an RSC request but we aren't in minimal mode, then we mark
-        // that this is a data request so that we can generate the flight data
-        // only.
-        if (!this.minimalMode) {
-          isDataReq = true
-        }
```

Comment:
This isn't needed anymore as we instead rely on the `isRSCRequest` and `isDynamicRSCRequest` to determine if we should return a data-only response.